prow/jobs/kyma-project/template-operator/template-operator.yaml

```diff
@@ -167,4 +163,100 @@ postsubmits: # runs on main
           - name: signify-secret
             secret:
               secretName: signify-dev-secret
-  
\ No newline at end of file
+    - name: release-template-operator-build
+      annotations:
+        description: "Job to build template-operator for a release."
+        owner: "jellyfish"
+      labels:
+        prow.k8s.io/pubsub.project: "sap-kyma-prow"
+        prow.k8s.io/pubsub.runID: "release-template-operator-build"
+        prow.k8s.io/pubsub.topic: "prowjobs"
+        preset-sa-kyma-push-images: "true"
+        preset-signify-prod-secret: "true"
+      always_run: true
+      skip_report: false
+      decorate: true
+      cluster: trusted-workload
+      max_concurrency: 10
+      branches:
+        - ^v?\d+\.\d+\.\d+(?:-.*)?$
+      spec:
+        containers:
+          - image: "eu.gcr.io/sap-kyma-neighbors-dev/image-builder:v20230313-8dfce5f0b-buildkit"
+            securityContext:
+              privileged: true
+              seccompProfile:
+                type: Unconfined
+              allowPrivilegeEscalation: true
+            command:
+              - "/image-builder"
+            args:
+              - "--name=template-operator"
+              - "--config=/config/kaniko-build-config.yaml"
+              - "--context=."
+              - "--dockerfile=components/operator/Dockerfile"
+              - "--tag=$(PULL_BASE_REF)"
+            env:
+              - name: BUILDKITD_FLAGS
+                value: "--oci-worker-no-process-sandbox"
+            resources:
+              requests:
+                memory: 1.5Gi
+                cpu: 1
+            volumeMounts:
+              - name: share
+                mountPath: /home/user/.local/share/buildkit
+              - name: config
+                mountPath: /config
+                readOnly: true
+              - name: signify-secret
+                mountPath: /secret
+                readOnly: true
+        volumes:
+          - name: share
+          - name: config
+            configMap:
+              name: kaniko-build-config
+          - name: signify-secret
+            secret:
+              secretName: signify-dev-secret
+    - name: release-template-operator-assets
+      annotations:
+        description: "Job to upload remaining template-operator assets to a release."
+        owner: "jellyfish"
+      labels:
+        prow.k8s.io/pubsub.project: "sap-kyma-prow"
+        prow.k8s.io/pubsub.runID: "release-template-operator-assets"
+        prow.k8s.io/pubsub.topic: "prowjobs"
+        preset-bot-github-token: "true"
+        preset-sa-kyma-push-images: "true"
+      always_run: true
+      skip_report: false
+      decorate: true
+      cluster: trusted-workload
+      max_concurrency: 10
+      branches:
+        - ^v?\d+\.\d+\.\d+(?:-.*)?$
```

Comment:
This will run even if there's no release, someone just creates a tag manually, like: `v1.2.3-test`. The referenced script handles that condition (running without a release), but it would be simpler if this could be a GH Action. Then we invoke the `upload_assets.sh` only when necessary.

Reply:
See above answer. Sure we can invest some time in the future to search for alternatives in GHA, but as of now I want to keep using prow jobs for this purpose and to finish the release issue within the sprint.

Reply:
@Ressetkk I like your sarcasm, but I can't really appreciate it enough. Perhaps you should redirect it to the people who are decisive in this matter? I just want to understand the reasons why things are setup in the specific way - I don't question or object any policy or technical contract. Quite the opposite - I want to understand how to conform to them.